msteams-platform/bots/how-to/conversations/conversation-messages.md

```diff
@@ -215,6 +223,312 @@ async def on_members_added_activity(
+    "eventType":"editMessage",
+    "tenant":{"id":"528dbe3f-15e0-4e37-84a1-00cc305847dd"}
+},
+"locale":"en-US",
+"localTimezone":"America/Los_Angeles"
+}  
+
+```
+
+# [Javascript](#tab/javascript3)
+
+You can use one of the following methods to get event notifications to handle the message updates using the Bot SDK:
+
+**​Event function registration**:
+
+```javascript
+
+this.onTeamsMessageEditEvent(async (context, next) => {
+  let editedMessage = context.activity.text;
+  let messageId = context.activity.id;
+  let previousMessage = this.currentMessage[messageId];
+
+  await context.sendActivity(`The previous message was "${previousMessage}". It is now: "${editedMessage}"`);
+
+  this.updateCurrentMessage(messageId, editedMessage);
+
+  next();
+})
+
+```
+
+**​Method override**:
+
+```javascript
+
+async onTeamsMessageEdit(context) {
+    let editedMessage = context.activity.text;
+    let messageId = context.activity.id;
+    let previousMessage = this.currentMessage[messageId];
+
+    await context.sendActivity(`The previous message was "${previousMessage}". It is now: "${editedMessage}"`);
+
+    this.updateCurrentMessage(messageId, editedMessage);
+}
+
+```
+
+---
+
+The following is an example of an undelete message event notification when a deleted message is restored:
+
+# [C#](#tab/csharp4)
+
+```csharp
+
+protected override async Task OnTeamsMessageUndeleteAsync(ITurnContext<IMessageUpdateActivity> turnContext, CancellationToken cancellationToken)
+{ 
+var replyActivity = MessageFactory.Text("message is undeleted"); 
+await turnContext.SendActivityAsync(replyActivity, cancellationToken); 
+} 
+
+```
+
+# [JSON](#tab/json4)
+
+```json
+
+{
+"type":"messageUpdate",
+"timestamp":"2022-10-28T17:19:39.4615413Z",
+"localTimestamp":"2022-10-28T10:19:39.4615413-07:00",
+"id":"1666977568748",
+"channelId":"msteams",
+"serviceUrl":"https://canary.botapi.skype.com/amer/",
+"from": {
+    "id":"29:1BLjP9j3_TM4mubmQZsYEo7jDyLeLf_YVA9sVPVO7KMAFMjJWB_EUGveb9EVDh9LgoNp9qjnzEBy4kgw83Jf1Kg",
+    "name":"Alex Wilber",
+    "aadObjectId":"976e4d1e-2108-43ee-a092-46a9507c5606"
+},
+"conversation":{
+    "conversationType":"personal",
+    "tenantId":"528dbe3f-15e0-4e37-84a1-00cc305847dd","id":"a:1tewuGJ44RkB90tiJNQ_I4q8vyuN5CYA_f-v6f0Vd-Bs3Ce85C73Ah1y8TvyjESsTHWjjgw-gnsuIuCUOWkfOCq6qaUYsk2_-fj93XXXHUMAUzhFFvTnaCU7V4WiMqXQL"
+},
+"recipient":{
+    "id":"28:0d469698-ab9d-479a-b0d8-758b6e6b1234",
+    "name":"Testbot"
+},
+"entities":[
+    {
+        "locale":"en-US",
+        "country":"US",
+        "platform":"Web",
+        "timezone":"America/Los_Angeles",
+        "type":"clientInfo"
+    }
+],
+"channelData":{
+    "eventType":"undeleteMessage",
+    "tenant":{"id":"528dbe3f-15e0-4e37-84a1-00cc305847dd"}
+},
+"locale":"en-US",
+"localTimezone":"America/Los_Angeles"
+}  
+
+```
+
+# [Javascript](#tab/javascript4)
+
+You can use one of the following methods to get event notifications to handle the message updates using the Bot SDK:
```

Comment:
Please do specify that use only one of the possible methods and NOT both. That is, if a developer is using the event function registration method to capture events, they should NOT use the method override method in the same class, and vice versa. This remark can also be found in the `TeamsActivityHandler.ts` class. https://github.com/microsoft/botbuilder-js/blob/6a23cfad98f11e65c23ba1dff0424d8a81329478/libraries/botbuilder/src/teamsActivityHandler.ts#L63-L68

Reply:
The same applies for all three of the newly added events.

Reply:
Updated as suggested.